v2/passwordless/serverless/with-aws-lambda/session-verification.mdx

```diff
@@ -11,57 +11,112 @@ import AppInfoForm from "/src/components/appInfoForm"
 import middy from "@middy/core";
 import cors from "@middy/http-cors";
 
 supertokens.init(getBackendConfig());
 
-//highlight-start
-const handler = async (event: SessionEvent) => {
+// @ts-ignore
+const lambdaHandler = async (event) => {
     return {
         body: JSON.stringify({
-            sessionHandle: event.session!.getHandle(),
-            userId: event.session!.getUserId(),
-            accessTokenPayload: event.session!.getAccessTokenPayload()
+            sessionHandle: event.session.getHandle(),
+            userId: event.session.getUserId(),
+            accessTokenPayload: event.session.getAccessTokenPayload(),
+        }),
+        statusCode: 200,
+    };
+};
+
+export const handler = middy(verifySession(lambdaHandler))
+    .use(
+        cors({
+            origin: getBackendConfig().appInfo.websiteDomain,
+            credentials: true,
+            headers: ["Content-Type", ...supertokens.getAllCORSHeaders()].join(", "),
+            methods: "OPTIONS,POST,GET,PUT,DELETE",
         })
-    }
-}
-
-module.exports.handler = middy(verifySession(handler)).use(cors({
-    origin: getBackendConfig().appInfo.websiteDomain,
-    credentials: true,
-    headers: ["Content-Type", ...supertokens.getAllCORSHeaders()].join(", "),
-    methods: "OPTIONS,POST,GET,PUT,DELETE"
-})).onError(request => {
-    throw request.error;
-});
-//highlight-end
+    )
+    .onError((request) => {
+        throw request.error;
+    });
 ```
 
+Now, import this function in your `index.mjs` handler file as shown below:
+
+
+```tsx title="index.mjs"
+import supertokens from "supertokens-node";
+import { middleware } from "supertokens-node/framework/awsLambda";
+// @ts-ignore
+import { getBackendConfig } from "./config.mjs";
+import middy from "@middy/core";
+import cors from "@middy/http-cors";
+// highlight-start
+// @ts-ignore
+import { handler as userHandler } from "./user.mjs";
+// highlight-end
+
+supertokens.init(getBackendConfig());
+
+export const handler = middy(
+    middleware((event) => {
+
+        // highlight-start
+        if (event.path === "/user") {
+            return userHandler(event);
+        // highlight-end
+        }
+      
+        return {
+            body: JSON.stringify({
+                msg: "Hello!",
+            }),
+            statusCode: 200,
+        };
+    })
+)
+    .use(
+        cors({
+            origin: getBackendConfig().appInfo.websiteDomain,
+            credentials: true,
+            headers: ["Content-Type", ...supertokens.getAllCORSHeaders()].join(", "),
+            methods: "OPTIONS,POST,GET,PUT,DELETE",
+        })
+    )
+    .onError((request) => {
+        throw request.error;
+    });
+```
+
+:::note
+The `verifySession` middleware automatically returns a 401 Unauthorised error if the session is not valid. You can alter the default behaviour by passing `{ sessionRequired: false }` as the second argument to the `verifySession` middleware.
+:::
```

Comment:
You can also mention that if you have a separate lambda function for the non auth routes, then you don';t need to use the superotkens middleware and can just use verifySession. You will have to do supertokens.init though, and provide the session recipe in the recipeList.

Reply:
I would avoid adding non-essential details in the docs. There are two cases - 

1. Someone forgets to call `SuperTokens.init` method then they would see an error message explaining the same.
2. Someone adds the auth middleware even when it wasn't required. Things would still work as expected.

Reply:
> Someone adds the auth middleware even when it wasn't required. Things would still work as expected.

I would avoid this cause it essentially exposes the APIs again. We should definitely add a note about this IMO.

Reply:
sure! I have done the changes.